experiment/selinux/src/selinux_label.rs

```diff
@@ -0,0 +1,305 @@
+use crate::selinux::*;
+use crate::tools::PathXattr;
+use crate::tools::*;
+use nix::sys::socket::getsockopt;
+use std::io::{BufRead, BufReader};
+use std::os::fd::AsFd;
+use std::path::Path;
+use std::sync::atomic::Ordering;
+
+const XATTR_NAME_SELINUX: &str = "security.selinux";
+const KEY_LABEL_PATH: &str = "/proc/self/attr/keycreate";
+
+// This impl is for methods related to labels in SELinux struct.
+impl SELinux {
+    // set_file_label sets the SELinux label for this path, following symlinks, or returns an error.
+    pub fn set_file_label<P: AsRef<Path> + PathXattr>(
+        fpath: P,
+        label: &str,
```

Comment:
Can't you use a struct to represent `label`? e.g., 
```rust
struct SELinuxLabel {
    user: String,
    role: String,
    type_: String,
    level: Option<String>,
}
```

Reply:
Thanks, I've updated the implementation. Could you review it again?